### PR TITLE
fix(plugins): fix useUserCameraDomElement behavior and did some refactor

### PR DESCRIPTION
**Pull request by GuiLeme**
### What does this PR do?

- Adds new extensible area: userCameraHelper (which is a button added in one of the corners of the user camera);
- Fixes useUserCameraDomElement (the issue was that when a new user enters the room, the dom-element-manipulator hook didn't return the elements);
- Adds a click event to the arguments of the onClick callback function of screenshareHelper;
- Refactors naming of plugin hook events so as to add new event informing bbb-core that the plugin is changing something on its side; 

### More

This PR is closely related to the SDK https://github.com/bigbluebutton/bigbluebutton-html-plugin-sdk/pull/110

See a demo of the new userCameraHelper down below:

image

### CODE REVIEW

**Review by Arthurk12 — CHANGES_REQUESTED** [review]

File: bigbluebutton-html5/imports/ui/components/video-provider/video-list/video-list-item/component.tsx [hunk 1]
```diff
@@ -60,12 +64,44 @@ interface VideoListItemProps {
...
+const renderPluginItems = (
+  streamId: string, userId: string,
+  pluginItems: UserCameraHelperInterface[],
+  bottom: boolean, right: boolean,
+) => {
+  if (pluginItems !== undefined) {
+    return (
+      <>
+        {
+          pluginItems.map((pluginItem) => {
```
Comment (germanocaumo) [inline bigbluebutton-html5/imports/ui/components/video-provider/video-list/video-list-item/component.tsx:76]: Missing the displayFunction filter here
```suggestion
          pluginItems.filter(
            (pluginItem) => (pluginItem.displayFunction?.({ userId, streamId }) ?? true),
          ).map((pluginItem) => {
```

File: bigbluebutton-html5/imports/ui/components/video-provider/video-list/video-list-item/component.tsx [hunk 2]
```diff
@@ -60,12 +64,44 @@ interface VideoListItemProps {
   };
 }
 
+const renderPluginItems = (
+  streamId: string, userId: string,
+  pluginItems: UserCameraHelperInterface[],
```
Comment (germanocaumo) [inline bigbluebutton-html5/imports/ui/components/video-provider/video-list/video-list-item/component.tsx:69]:
```suggestion
  pluginItems: UserCameraHelperButton[],
```

**Review by germanocaumo — CHANGES_REQUESTED** [review]
Missing displayFunction in UserCameraHelperItems

**Review by Arthurk12 — APPROVED** [review]
LGTM.

**Pull request merged**